File: fresh-shop/server/router/shop/order.go
package shop

import (
	"fresh-shop/server/api/v1"
	"fresh-shop/server/middleware"
	"github.com/gin-gonic/gin"
)

type OrderRouter struct {
}

// InitOrderRouter 初始化 Order 路由信息
func (s *OrderRouter) InitOrderRouter(Router *gin.RouterGroup) {
	orderRouter := Router.Group("order").Use(middleware.OperationRecord())
	orderRouterWithoutRecord := Router.Group("order")
	var orderApi = v1.ApiGroupApp.ShopApiGroup.OrderApi
	{
		orderRouter.POST("createOrder", orderApi.CreateOrder)             // 创建待支付 Order
		orderRouter.POST("orderPay", orderApi.OrderPay)                   // 支付 Order, 返回微信支付所需要的参数
		orderRouter.POST("cancelOrder", orderApi.CancelOrder)             // 取消订单
		orderRouter.DELETE("deleteOrder", orderApi.DeleteOrder)           // 删除 Order
		orderRouter.DELETE("deleteOrderByIds", orderApi.DeleteOrderByIds) // 批量删除 Order
		orderRouter.PUT("updateOrder", orderApi.UpdateOrder)              // 更新 Order
	}
	{
		orderRouterWithoutRecord.GET("findOrder", orderApi.FindOrder)                     // 根据ID获取Order
		orderRouterWithoutRecord.GET("findUserOrderStatus", orderApi.FindUserOrderStatus) // 获取用户订单中数量
		orderRouterWithoutRecord.GET("getOrderList", orderApi.GetOrderList)               // 获取Order列表
		orderRouterWithoutRecord.GET("getUserOrderList", orderApi.GetUserOrderList)       // 根据登录用户获取Order列表
		orderRouterWithoutRecord.GET("orderStatus", orderApi.OrderStatus)                 // 获取订单状态 Order
	}
}

File: fresh-shop-uniapp/api/order.js
import request from "@/utils/request"

// 订单支付
export const orderPay = (data, refs) => {
    return request({
        url: `/order/orderPay`,
        method: 'POST',
        loading: true,
        data
    }, refs)
}

// 获取默认列表
export const getOrderList = (data, loading, refs) => {
    return request({
        url: `/order/getUserOrderList`,
        method: 'GET',
        loading: loading,
        toLogin: true,
        data
    }, refs)
}

// 获取订单
export const getOrderInfo = (data, refs) => {
    return request({
        url: `/order/findOrder`,
        method: 'GET',
        loading: true,
        data
    }, refs)
}

// 用户订单数量统计
export const getOrderStatusCount = (data, refs) => {
    return request({
        url: `/order/findUserOrderStatus`,
        method: 'GET',
        loading: false,
        data
    }, refs)
}

// 获取订单状态
export const getOrderStatus = (orderId, refs) => {
    return request({
        url: `/order/orderStatus`,
        method: 'GET',
        loading: false,
        data: {
            ID: orderId
        }
    }, refs)
}

// 创建订单
export const createOrder = (data, refs) => {
    return request({
        url: `/order/createOrder`,
        method: 'POST',
        loading: true,
        data
    }, refs)
}

// 更新订单
export const updateOrder = (data, refs) => {
    return request({
        url: `/order/updateOrder`,
        method: 'PUT',
        loading: true,
        data
    }, refs)
}

// 更新订单
export const deleteOrder = (data, refs) => {
    return request({
        url: `/order/deleteOrder`,
        method: 'DELETE',
        loading: true,
        data
    }, refs)
}

// 取消订单
export const cancelOrder = (data, refs) => {
    return request({
        url: `/order/cancelOrder`,
        method: 'POST',
        loading: true,
        data
    }, refs)
}


// 确认收货
export const confirmOrder = (data, refs) => {
    return request({
        url: `/order/confirmOrder`,
        method: 'POST',
        loading: true,
        data
    }, refs)
}

